src/iteration_utils.jl

```diff
@@ -22,84 +22,85 @@ _use_findstructralnz(::SparseMatrixCSC) = false
 
 function __init__()
     @require BlockBandedMatrices="ffab5731-97b5-5995-9138-79e8c1846df0" begin
-        _use_findstructralnz(::BlockBandedMatrices.BandedBlockBandedMatrix) = false
-        _use_findstructralnz(::BlockBandedMatrices.BlockBandedMatrix) = false
+        @require BlockArrays="8e7c35d0-a365-5155-bbbb-fb81a777f24e" begin
+            _use_findstructralnz(::BlockBandedMatrices.BandedBlockBandedMatrix) = false
+            _use_findstructralnz(::BlockBandedMatrices.BlockBandedMatrix) = false
 
-        @inline function _colorediteration!(Jac::BlockBandedMatrices.BandedBlockBandedMatrix,
-                                            sparsity::BlockBandedMatrices.BandedBlockBandedMatrix,
-                                            rows_index,cols_index,vfx,colorvec,color_i,ncols)
-            λ,μ = BlockBandedMatrices.subblockbandwidths(Jac)
-            rs = BlockBandedMatrices.BlockSizes((BlockBandedMatrices.cumulsizes(Jac,1),)) # column block sizes
-            cs = BlockBandedMatrices.BlockSizes((BlockBandedMatrices.cumulsizes(Jac,2),))
-            b = BlockBandedMatrices.BlockArray(vfx,rs)
-            c = BlockBandedMatrices.BlockArray(colorvec,cs)
-            @inbounds for J=BlockBandedMatrices.Block.(1:BlockBandedMatrices.nblocks(Jac,2))
-                c_v = c.blocks[J.n[1]]
-                @inbounds for K=BlockBandedMatrices.blockcolrange(Jac,J)
-                    V = view(Jac,K,J)
-                    b_v = b.blocks[K.n[1]]
-                    data = BlockBandedMatrices.bandeddata(V)
-                    p = pointer(data)
-                    st = stride(data,2)
-                    m,n = size(V)
-                    @inbounds for j=1:n
-                        if c_v[j] == color_i
-                            @inbounds for k=max(1,j-μ):min(m,j+λ)
-                                unsafe_store!(p, b_v[k], (j-1)*st + μ + k - j + 1)
+            @inline function _colorediteration!(Jac::BlockBandedMatrices.BandedBlockBandedMatrix,
+                                                sparsity::BlockBandedMatrices.BandedBlockBandedMatrix,
+                                                rows_index,cols_index,vfx,colorvec,color_i,ncols)
+                λ,μ = BlockBandedMatrices.subblockbandwidths(Jac)
+                rs = BlockArrays.blocklasts(BlockArrays.axes(Jac,1)) # column block sizes

```

Comment:
Probably should be ` blocklengths(axes(Jac,1)) ` to get the lengths of the blocks

Reply:
works